**"Democracy and Social Development: A Data-Driven Analysis"**


By Raphael J.Akeyo, May 2024

Data Analyst, Spiced Academy
Berlin, Germany

**Research Question**

This study seeks to answer the question: How Does Democratic Governance influence Social Development? In order to unpack how various tenets of democracy influence social development outcomes in different countries, the main research question will be guided by the following specific questions;   

The specific research questions are:

1. What is the impact of democratic governance on social development?
2. How does participation as a central tenet of democracy impact social development indicators? 
3. How does justice and the rule of law impact social development? 
4. To what extent does democratic freedoms correlate with social development indicators?
5. How does accountability (measured through coruption indices) influence social development?    

**Data**

1. Data:   The Global State of Democracy Indices
   
   Source: The International Institute for Democracy and Electoral Assistance (International IDEA). (2023). The Global State of Democracy Indices, 1975–2022, v. 7, 2023, <https://www.idea.int/gsod-indices>, (Accesed on 8th May 2024).
   

   

**Limitations of the study**
1. The period of data 1975 to 2020. The study only focuses on the period between 1975 to 2022. Reason: this is the period of the commencement and progress of democracy from its third global wave which started in 1974 upto the latest available data as at the date of this study (Huntington, 1991). 
2. Even though the democracy index is calculated/ measured based on several indicators around the main tenets of; participation, representation, rights and the rule of law (International IDEA, 2023), this study has meticulously sampled a number of variables relevant to this research question. It must not therefore be assumed that only the variables used in this study make up the overall democracy index. Conversely, it must also be noted that the sampling done in this study does not any any way undermine the reliability of the results as it only asseses the impact of specific variables(as sampled) on specific social development indices (as sampled). Further studies are recommended for a more comprehensive approach focusing on all or any other aspects of the many variables of democracy and development. 

**The datasets: A Sneak-peek into the data?**


In [1]:
import pandas as pd

In [2]:
#Global State of Democracy Index DataSet;

df_global_democracyIndex = pd.read_excel('Global_state_of_democracy_index.xlsx')
df_global_democracyIndex

ID_country_name  ID_country_code  ID_year  ID_country_year  ID_region  \
0      United States                2     1975            21975        3.0   
1      United States                2     1976            21976        3.0   
2      United States                2     1977            21977        3.0   
3      United States                2     1978            21978        3.0   
4      United States                2     1979            21979        3.0   
...              ...              ...      ...              ...        ...   
8528           World              999     2016          9992016        NaN   
8529           World              999     2017          9992017        NaN   
8530           World              999     2018          9992018        NaN   
8531           World              999     2019          9992019        NaN   
8532           World              999     2020          9992020        NaN   

      ID_subregion ID_region_name ID_subregion_name      C_A1      L_A1  ...  \
0              9.0  North America     North America  0.739810  0.677210  ...   
1              9.0  North America     North America  0.753778  0.689358  ...   
2              9.0  North America     North America  0.822535  0.758995  ...   
3              9.0  North America     North America  0.821670  0.757696  ...   
4              9.0  North America     North America  0.823376  0.758889  ...   
...            ...            ...               ...       ...       ...  ...   
8528           NaN            NaN               NaN  0.563625  0.499823  ...   
8529           NaN            NaN               NaN  0.562800  0.498907  ...   
8530           NaN            NaN               NaN  0.558651  0.494782  ...   
8531           NaN            NaN               NaN  0.553332  0.489436  ...   
8532           NaN            NaN               NaN  0.552947  0.489178  ...   

       v_51_05   v_51_06  v_52_01  v_53_01  v_53_02   v_54_01   v_54_02  \
0     0.614308  0.714905   0.3823      0.0      1.0  0.998996  0.779491   
1     0.614308  0.714905   0.5692      0.0      1.0  0.998996  0.779491   
2     0.614308  0.714905   0.5692      0.0      1.0  0.998996  0.779491   
3     0.614308  0.674620   0.3720      0.0      1.0  0.998996  0.779491   
4     0.614308  0.715525   0.3720      0.0      1.0  0.998996  0.779491   
...        ...       ...      ...      ...      ...       ...       ...   
8528       NaN       NaN      NaN      NaN      NaN       NaN       NaN   
8529       NaN       NaN      NaN      NaN      NaN       NaN       NaN   
8530       NaN       NaN      NaN      NaN      NaN       NaN       NaN   
8531       NaN       NaN      NaN      NaN      NaN       NaN       NaN   
8532       NaN       NaN      NaN      NaN      NaN       NaN       NaN   

      regime_status_name  democratic_performance_name  \
0              Democracy    High performing democracy   
1              Democracy    High performing democracy   
2              Democracy    High performing democracy   
3              Democracy    High performing democracy   
4              Democracy    High performing democracy   
...                  ...                          ...   
8528                 NaN                          NaN   
8529                 NaN                          NaN   
8530                 NaN                          NaN   
8531                 NaN                          NaN   
8532                 NaN                          NaN   

      democratic_performance_numeric  
0                                1.0  
1                                1.0  
2                                1.0  
3                                1.0  
4                                1.0  
...                              ...  
8528                             NaN  
8529                             NaN  
8530                             NaN  
8531                             NaN  
8532                             NaN  

[8533 rows x 203 columns]

In [3]:
df_global_democracyIndex.columns

Index(['ID_country_name', 'ID_country_code', 'ID_year', 'ID_country_year',
       'ID_region', 'ID_subregion', 'ID_region_name', 'ID_subregion_name',
       'C_A1', 'L_A1',
       ...
       'v_51_05', 'v_51_06', 'v_52_01', 'v_53_01', 'v_53_02', 'v_54_01',
       'v_54_02', 'regime_status_name', 'democratic_performance_name',
       'democratic_performance_numeric'],
      dtype='object', length=203)

In [4]:
df_global_democracyIndex['ID_country_name'].unique()

array(['United States', 'Canada', 'Cuba', 'Haiti', 'Dominican Republic',
       'Jamaica', 'Trinidad and Tobago', 'Barbados', 'Mexico',
       'Guatemala', 'Honduras', 'El Salvador', 'Nicaragua', 'Costa Rica',
       'Panama', 'Colombia', 'Venezuela', 'Ecuador', 'Peru', 'Brazil',
       'Bolivia', 'Paraguay', 'Chile', 'Argentina', 'Uruguay',
       'United Kingdom', 'Ireland', 'Netherlands', 'Belgium',
       'Luxembourg', 'France', 'Switzerland', 'Spain', 'Portugal',
       'Germany', 'East Germany', 'Poland', 'Austria', 'Hungary',
       'Czechia', 'Slovakia', 'Italy', 'Albania', 'Montenegro',
       'North Macedonia', 'Croatia', 'Serbia', 'Bosnia and Herzegovina',
       'Kosovo', 'Slovenia', 'Greece', 'Cyprus', 'Bulgaria', 'Moldova',
       'Romania', 'Russia', 'Estonia', 'Latvia', 'Lithuania', 'Ukraine',
       'Belarus', 'Armenia', 'Georgia', 'Azerbaijan', 'Finland', 'Sweden',
       'Norway', 'Denmark', 'Iceland', 'Cape Verde', 'Guinea-Bissau',
       'Equatorial Guinea', 'Gambi

In [5]:
#Counting the number of countries in the dataset as listed above;
number_of_countries = df_global_democracyIndex['ID_country_name'].nunique()
print(f'The number of countries in the dataset is: {number_of_countries}')


The number of countries in the dataset is: 195


The dataset has a total of 8533 rows and 203 colums with data on 195 Countries as listed above. 

*Selection of Useful Columns*

From the 203 columns in the dataset above, I will only select useful columns inline with the variables relevant to this research question. In this study, I conceptualize democracy as afunction of the following central tenets; Representation, Rights, Rule of Law and Participation. These tenets are measured using various indicators as adopted from the following credible authorities; 

1. Claudiu D. Tufis, Alexander Hudson, (2023). The Global State of Democracy Indices Codebook Version 7 . International IDEA Strömsborg SE–103 34 Stockholm SWEDEN
2. International IDEA, The Global State of Democracy Indices, 1975–2022, v. 7, 2023, <https://www.idea.int/gsod-indices>, [14th May 2024]

The selected columns are therefore useful indicators used to measure various tenets of democracy and shall be adopted as the variables of measurement in this study. 


In [6]:
#List of selected. The column names are coded using the Global State of Democracy Indices Codebook Version 7 (2023) and shall be decoded appropriately and the columns renamed.
columns_to_keep = [
    'ID_country_name', 'ID_country_code', 'ID_year', 'ID_region_name',
    'regime_status_name', 'democratic_performance_name', 'democratic_performance_numeric',
    'v_41_05', 'v_41_04', 'v_33_02', 'v_33_01', 'v_32_06', 'v_32_05', 'v_32_01',
    'v_31_05', 'v_31_04', 'v_23_07', 'v_23_06', 'v_23_04', 'v_23_02', 'v_23_01',
    'v_22_43', 'v_22_42', 'v_22_41', 'v_22_32', 'v_22_31', 'v_22_24', 'v_22_22',
    'v_22_15', 'v_22_13', 'v_22_06', 'v_22_05', 'v_22_03', 'v_22_02', 'v_22_01',
    'v_21_05', 'v_21_04', 'v_21_03', 'v_21_02', 'v_21_01', 'v_13_06', 'v_12_01',
    'v_11_06', 'v_11_05', 'v_11_01', 'v_11_02'
]

#Selecting only the selected columns
df_democracyIndex_selected_columns = df_global_democracyIndex[columns_to_keep]
df_democracyIndex_selected_columns.columns

Index(['ID_country_name', 'ID_country_code', 'ID_year', 'ID_region_name',
       'regime_status_name', 'democratic_performance_name',
       'democratic_performance_numeric', 'v_41_05', 'v_41_04', 'v_33_02',
       'v_33_01', 'v_32_06', 'v_32_05', 'v_32_01', 'v_31_05', 'v_31_04',
       'v_23_07', 'v_23_06', 'v_23_04', 'v_23_02', 'v_23_01', 'v_22_43',
       'v_22_42', 'v_22_41', 'v_22_32', 'v_22_31', 'v_22_24', 'v_22_22',
       'v_22_15', 'v_22_13', 'v_22_06', 'v_22_05', 'v_22_03', 'v_22_02',
       'v_22_01', 'v_21_05', 'v_21_04', 'v_21_03', 'v_21_02', 'v_21_01',
       'v_13_06', 'v_12_01', 'v_11_06', 'v_11_05', 'v_11_01', 'v_11_02'],
      dtype='object')

Only 46 columns are left and shall be used in this study. The columns are as displayed above

*Renaming the columns to depict the actual indicators measured*

In [7]:
df_democracyIndex_selected_columns.rename(columns={'ID_country_name': 'country', 'ID_country_code': 'country_code', 
                                                   'ID_year':'year', 'ID_region_name':'region', 
                                                   'regime_status_name': 'regime', 
                                                   'democratic_performance_name': 'democratic_performance',
                                                  'democratic_performance_numeric':'democracy_index', 
                                                  'v_41_05': 'participation_interest_grps', 'v_41_04':'e_participation',
                                                  'v_33_02':'transparent_laws', 'v_33_01':'constitution_respect',
                                                  'v_32_06':'functioning_gvnt', 'v_32_05':'corruption', 'v_32_01':'public_sector_corruption',
                                                  'v_31_05':'judiciary_independence', 'v_31_04':'judicial_compliance', 
                                                   'v_23_07':'health_equality', 'v_23_06':'education_equality', 'v_23_04':'literacy_index',
                                                  'v_23_02':'life_expectancy', 'v_23_01':'infant_mortality', 
                                                   'v_22_43':'freedom_of_domestic_movement_for_men', 
                                                   'v_22_42':'freedom_of_domestic_movement_for_women',
                                                  'v_22_41':'freedom_of_foreign_movement', 'v_22_32':'religious_repression',
                                                  'v_22_31':'religious_freedom', 'v_22_24':'freedom_assemblyandassociation',
                                                  'v_22_22':'civil_society_repression', 'v_22_15':'freedom_of_critisms_by_media', 
                                                   'v_22_13':'journalists_harrassment', 'v_22_06':'infrastructural_challenges_to_access_gvntservices',
                                                  'v_22_05':'speech_and_press_freedom', 'v_22_03':'academic_and_cultural_freedom',
                                                  'v_22_02':'freedom_of_discusion_men', 'v_22_01':'freedom_of_discusion_women', 
                                                  'v_21_05':'judicial_fair_trial', 'v_21_04':'judicial_accountability', 'v_21_03':'judicial_corruption',
                                                  'v_21_02':'justice_access_women', 'v_21_01':'justice_access_men',
                                                  'v_13_06':'multiparty_elections', 'v_12_01':'inclusive_suffrage', 'v_11_06':'competetive_elections',
                                                  'v_11_05':'free_fair_elections', 'v_11_01':'autonomy_of_electoral_body', 
                                                   'v_11_02':'capacity_of_electoral_body' }, inplace=True)

C:\Users\Raphael Akeyo\AppData\Local\Temp\ipykernel_17932\1353930158.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_democracyIndex_selected_columns.rename(columns={'ID_country_name': 'country', 'ID_country_code': 'country_code',


In [8]:
df_democracyIndex_selected_columns.columns

Index(['country', 'country_code', 'year', 'region', 'regime',
       'democratic_performance', 'democracy_index',
       'participation_interest_grps', 'e_participation', 'transparent_laws',
       'constitution_respect', 'functioning_gvnt', 'corruption',
       'public_sector_corruption', 'judiciary_independence',
       'judicial_compliance', 'health_equality', 'education_equality',
       'literacy_index', 'life_expectancy', 'infant_mortality',
       'freedom_of_domestic_movement_for_men',
       'freedom_of_domestic_movement_for_women', 'freedom_of_foreign_movement',
       'religious_repression', 'religious_freedom',
       'freedom_assemblyandassociation', 'civil_society_repression',
       'freedom_of_critisms_by_media', 'journalists_harrassment',
       'infrastructural_challenges_to_access_gvntservices',
       'speech_and_press_freedom', 'academic_and_cultural_freedom',
       'freedom_of_discusion_men', 'freedom_of_discusion_women',
       'judicial_fair_trial', 'judicial_a

In [9]:
df_democracyIndex_selected_columns.isnull().sum()

country                                                 0
country_code                                            0
year                                                    0
region                                                276
regime                                               1431
democratic_performance                               1431
democracy_index                                      1431
participation_interest_grps                          3810
e_participation                                      1410
transparent_laws                                     1410
constitution_respect                                 1410
functioning_gvnt                                     2433
corruption                                           3810
public_sector_corruption                             1437
judiciary_independence                               1832
judicial_compliance                                  1525
health_equality                                      1483
education_equa

N/B: Most of the null values are associated with regional metrices such as; Africa, Europe, World, Asia, etc and can be dropped as below;

In [10]:
#List of rows to drop based on 'country'
rows_to_drop = [
    'East Germany', 'African Union', 'ASEAN', 'European Union', 'OECD', 'OAS', 'East Africa',
    'Central Africa', 'Southern Africa', 'West Africa', 'North Africa', 'Caribbean', 'Central America',
    'South America', 'North America', 'Central Asia', 'East Asia', 'South Asia', 'South-East Asia',
    'Oceania', 'Middle East', 'East-Central Europe', 'Eastern Europe', 'North/Western Europe', 'Southern Europe',
    'Africa', 'Latin America/Caribbean', 'Asia/Pacific', 'Europe', 'World'
]

df_democracyIndex_filtered_rows = df_democracyIndex_selected_columns[~df_democracyIndex_selected_columns['country'].isin(rows_to_drop)]



In [11]:
#checking for new null values after dropping less-useful rows;
df_democracyIndex_filtered_rows.isnull().sum()

country                                                 0
country_code                                            0
year                                                    0
region                                                  0
regime                                                 21
democratic_performance                                 21
democracy_index                                        21
participation_interest_grps                          2391
e_participation                                         0
transparent_laws                                        0
constitution_respect                                    0
functioning_gvnt                                     1007
corruption                                           2391
public_sector_corruption                               27
judiciary_independence                                420
judicial_compliance                                   115
health_equality                                        57
education_equa

*How to handle the null values:*

1. I will group by country and calculate mean for all numeric columns the use the mean to fill the missing values in the numeric columns
2. For the non-numeric columns, I will fill using the mode.
3. No null values will be dropped. 

In [18]:
#Grouping by country and calculating the mean for numeric columns only;

# List of numeric columns excluding 'year' and 'country_code'
numeric_columns = [
    'democracy_index', 'participation_interest_grps', 
    'e_participation', 'transparent_laws', 'constitution_respect', 
    'functioning_gvnt', 'corruption', 'public_sector_corruption', 
    'judiciary_independence', 'judicial_compliance', 'health_equality', 
    'education_equality', 'literacy_index', 'life_expectancy', 'infant_mortality',
    'freedom_of_domestic_movement_for_men', 'freedom_of_domestic_movement_for_women', 
    'freedom_of_foreign_movement', 'religious_repression', 'religious_freedom',
    'freedom_assemblyandassociation', 'civil_society_repression', 
    'freedom_of_critisms_by_media', 'journalists_harrassment',
    'infrastructural_challenges_to_access_gvntservices', 'speech_and_press_freedom', 
    'academic_and_cultural_freedom', 'freedom_of_discusion_men', 
    'freedom_of_discusion_women', 'judicial_fair_trial', 'judicial_accountability', 
    'judicial_corruption', 'justice_access_women', 'justice_access_men', 
    'multiparty_elections', 'inclusive_suffrage', 'competetive_elections', 
    'free_fair_elections', 'autonomy_of_electoral_body', 'capacity_of_electoral_body'
]

# Calculating the mean for each numeric column;
column_means = df_democracyIndex_filtered_rows[numeric_columns].mean()

# Filling in the missing values in the numeric columns with the mean of each numeric column
df_filled_numeric = df_democracyIndex_filtered_rows.copy()
df_filled_numeric[numeric_columns] = df_democracyIndex_filtered_rows[numeric_columns].fillna(column_means)

df_filled_numeric

country  country_code  year         region     regime  \
0     United States             2  1975  North America  Democracy   
1     United States             2  1976  North America  Democracy   
2     United States             2  1977  North America  Democracy   
3     United States             2  1978  North America  Democracy   
4     United States             2  1979  North America  Democracy   
...             ...           ...   ...            ...        ...   
7118           Fiji           950  2016   Asia/Pacific  Democracy   
7119           Fiji           950  2017   Asia/Pacific  Democracy   
7120           Fiji           950  2018   Asia/Pacific  Democracy   
7121           Fiji           950  2019   Asia/Pacific  Democracy   
7122           Fiji           950  2020   Asia/Pacific  Democracy   

         democratic_performance  democracy_index  participation_interest_grps  \
0     High performing democracy              1.0                     0.486684   
1     High performing democracy              1.0                     0.486684   
2     High performing democracy              1.0                     0.486684   
3     High performing democracy              1.0                     0.486684   
4     High performing democracy              1.0                     0.486684   
...                         ...              ...                          ...   
7118             Weak democracy              3.0                     0.486684   
7119             Weak democracy              3.0                     0.486684   
7120             Weak democracy              3.0                     0.486684   
7121             Weak democracy              3.0                     0.486684   
7122             Weak democracy              3.0                     0.486684   

      e_participation  transparent_laws  ...  judicial_accountability  \
0            0.772996          0.743978  ...                 0.825735   
1            0.772996          0.743978  ...                 0.825735   
2            0.772996          0.743978  ...                 0.825735   
3            0.772996          0.743978  ...                 0.825735   
4            0.772996          0.743978  ...                 0.825735   
...               ...               ...  ...                      ...   
7118         0.536173          0.601145  ...                 0.732010   
7119         0.552347          0.601145  ...                 0.682764   
7120         0.552347          0.601145  ...                 0.656871   
7121         0.552347          0.601145  ...                 0.671009   
7122         0.583538          0.575880  ...                 0.641779   

      judicial_corruption  justice_access_women  justice_access_men  \
0                0.737792              0.788124            0.834769   
1                0.737792              0.788124            0.834769   
2                0.737792              0.788124            0.834769   
3                0.737792              0.788124            0.834769   
4                0.737792              0.788124            0.834769   
...                   ...                   ...                 ...   
7118             0.710583              0.606131            0.633426   
7119             0.705427              0.637938            0.652306   
7120             0.685522              0.610619            0.652306   
7121             0.685522              0.610619            0.652306   
7122             0.685522              0.601898            0.628231   

      multiparty_elections  inclusive_suffrage  competetive_elections  \
0                      1.0                 1.0                    1.0   
1                      1.0                 1.0                    1.0   
2                      1.0                 1.0                    1.0   
3                      1.0                 1.0                    1.0   
4                      1.0                 1.0                    1.0   
...                    ...                

In [19]:
#checking null values after filling numeric columns;
df_filled_numeric.isnull().sum()

country                                               0
country_code                                          0
year                                                  0
region                                                0
regime                                               21
democratic_performance                               21
democracy_index                                       0
participation_interest_grps                           0
e_participation                                       0
transparent_laws                                      0
constitution_respect                                  0
functioning_gvnt                                      0
corruption                                            0
public_sector_corruption                              0
judiciary_independence                                0
judicial_compliance                                   0
health_equality                                       0
education_equality                              

There are 21 countries without classifications for 'democratic performance' and 'regime'. We shall use mode to fill in these missing values. Justification is that, the most occuring classification for that country will most likely suffice for the missing values under 'regime' and 'category of democratic performance'. 

In [20]:

# List of non-numeric columns with missing values
non_numeric_columns = ['regime', 'democratic_performance']

# Group the data by 'country' and calculate the mode for non-numeric columns
mode_per_country = df_filled_numeric.groupby('country')[non_numeric_columns].transform(lambda x: x.mode().iloc[0])

# Fill in missing values with the mode per country
df_filled = df_filled_numeric.copy()
df_filled[non_numeric_columns] = df_filled_numeric[non_numeric_columns].fillna(mode_per_country)

df_filled

country  country_code  year         region     regime  \
0     United States             2  1975  North America  Democracy   
1     United States             2  1976  North America  Democracy   
2     United States             2  1977  North America  Democracy   
3     United States             2  1978  North America  Democracy   
4     United States             2  1979  North America  Democracy   
...             ...           ...   ...            ...        ...   
7118           Fiji           950  2016   Asia/Pacific  Democracy   
7119           Fiji           950  2017   Asia/Pacific  Democracy   
7120           Fiji           950  2018   Asia/Pacific  Democracy   
7121           Fiji           950  2019   Asia/Pacific  Democracy   
7122           Fiji           950  2020   Asia/Pacific  Democracy   

         democratic_performance  democracy_index  participation_interest_grps  \
0     High performing democracy              1.0                     0.486684   
1     High performing democracy              1.0                     0.486684   
2     High performing democracy              1.0                     0.486684   
3     High performing democracy              1.0                     0.486684   
4     High performing democracy              1.0                     0.486684   
...                         ...              ...                          ...   
7118             Weak democracy              3.0                     0.486684   
7119             Weak democracy              3.0                     0.486684   
7120             Weak democracy              3.0                     0.486684   
7121             Weak democracy              3.0                     0.486684   
7122             Weak democracy              3.0                     0.486684   

      e_participation  transparent_laws  ...  judicial_accountability  \
0            0.772996          0.743978  ...                 0.825735   
1            0.772996          0.743978  ...                 0.825735   
2            0.772996          0.743978  ...                 0.825735   
3            0.772996          0.743978  ...                 0.825735   
4            0.772996          0.743978  ...                 0.825735   
...               ...               ...  ...                      ...   
7118         0.536173          0.601145  ...                 0.732010   
7119         0.552347          0.601145  ...                 0.682764   
7120         0.552347          0.601145  ...                 0.656871   
7121         0.552347          0.601145  ...                 0.671009   
7122         0.583538          0.575880  ...                 0.641779   

      judicial_corruption  justice_access_women  justice_access_men  \
0                0.737792              0.788124            0.834769   
1                0.737792              0.788124            0.834769   
2                0.737792              0.788124            0.834769   
3                0.737792              0.788124            0.834769   
4                0.737792              0.788124            0.834769   
...                   ...                   ...                 ...   
7118             0.710583              0.606131            0.633426   
7119             0.705427              0.637938            0.652306   
7120             0.685522              0.610619            0.652306   
7121             0.685522              0.610619            0.652306   
7122             0.685522              0.601898            0.628231   

      multiparty_elections  inclusive_suffrage  competetive_elections  \
0                      1.0                 1.0                    1.0   
1                      1.0                 1.0                    1.0   
2                      1.0                 1.0                    1.0   
3                      1.0                 1.0                    1.0   
4                      1.0                 1.0                    1.0   
...                    ...                

In [21]:
#checking null values in the entire dataframe;
df_filled.isnull().sum()

country                                              0
country_code                                         0
year                                                 0
region                                               0
regime                                               0
democratic_performance                               0
democracy_index                                      0
participation_interest_grps                          0
e_participation                                      0
transparent_laws                                     0
constitution_respect                                 0
functioning_gvnt                                     0
corruption                                           0
public_sector_corruption                             0
judiciary_independence                               0
judicial_compliance                                  0
health_equality                                      0
education_equality                                   0
literacy_i

**The Data is clean. Now I save the clean dataframe as 'df_democracyIndex_clean' for exporting to tableau for further analysis.**

In [22]:
#saving the clean dataframe;
file_path = 'df_democracyIndex_clean.csv'

# Saving as a CSV file
df_filled.to_csv(file_path, index=False)

print("DataFrame saved to:", file_path)   #just to confirm succesfull saving

DataFrame saved to: df_democracyIndex_clean.csv


In [24]:
#displaying the last 5 rows of the clean data to just confirm;
df_democracyIndex_clean=pd.read_csv('df_democracyIndex_clean.csv')
df_democracyIndex_clean.tail(5)

country  country_code  year        region     regime  \
7102    Fiji           950  2016  Asia/Pacific  Democracy   
7103    Fiji           950  2017  Asia/Pacific  Democracy   
7104    Fiji           950  2018  Asia/Pacific  Democracy   
7105    Fiji           950  2019  Asia/Pacific  Democracy   
7106    Fiji           950  2020  Asia/Pacific  Democracy   

     democratic_performance  democracy_index  participation_interest_grps  \
7102         Weak democracy              3.0                     0.486684   
7103         Weak democracy              3.0                     0.486684   
7104         Weak democracy              3.0                     0.486684   
7105         Weak democracy              3.0                     0.486684   
7106         Weak democracy              3.0                     0.486684   

      e_participation  transparent_laws  ...  judicial_accountability  \
7102         0.536173          0.601145  ...                 0.732010   
7103         0.552347          0.601145  ...                 0.682764   
7104         0.552347          0.601145  ...                 0.656871   
7105         0.552347          0.601145  ...                 0.671009   
7106         0.583538          0.575880  ...                 0.641779   

      judicial_corruption  justice_access_women  justice_access_men  \
7102             0.710583              0.606131            0.633426   
7103             0.705427              0.637938            0.652306   
7104             0.685522              0.610619            0.652306   
7105             0.685522              0.610619            0.652306   
7106             0.685522              0.601898            0.628231   

      multiparty_elections  inclusive_suffrage  competetive_elections  \
7102                   1.0                 1.0                    1.0   
7103                   1.0                 1.0                    1.0   
7104                   1.0                 1.0                    1.0   
7105                   1.0                 1.0                    1.0   
7106                   1.0                 1.0                    1.0   

      free_fair_elections  autonomy_of_electoral_body  \
7102             0.541316                    0.434723   
7103             0.541316                    0.434723   
7104             0.490279                    0.418786   
7105             0.490279                    0.418786   
7106             0.490279                    0.505516   

      capacity_of_electoral_body  
7102                    0.645966  
7103                    0.678208  
7104                    0.646601  
7105                    0.607211  
7106                    0.646442  

[5 rows x 46 columns]

**Explaining the Variables under Study**

The data presents two sets of variables; democracy variables and social development variables as discussed below; 

**Democratic Variables**: These variables relate to the functioning and performance of democratic institutions and practices:

1. Democratic Performance: General performance of democracy.
2. Democracy Index: Overall index measuring democracy.
3. Participation Interest Groups: Engagement of interest groups in the democratic process.
4. E-Participation: Digital participation in democratic processes.
5. Transparent Laws: Transparency in law-making processes.
6. Constitution Respect: Respect for constitutional provisions.
7. Functioning Government: Effectiveness of government functioning.
8. Corruption: Levels of corruption in the country.
9. Public Sector Corruption: Corruption specifically within the public sector.
10. Judiciary Independence: Independence of the judiciary.
11. Judicial Compliance: Compliance of judiciary with democratic norms.
12. Freedom of Domestic Movement for Men/Women: Freedom of movement within the country.
13. Freedom of Foreign Movement: Freedom to travel abroad.
14. Religious Repression/Freedom: Levels of religious repression or freedom.
15. Freedom of Assembly and Association: Ability to freely assemble and associate.
16. Civil Society Repression: Repression of civil society activities.
17. Freedom of Criticism by Media: Freedom for media to criticize the government.
18. Journalists Harassment: Harassment faced by journalists.
19. Speech and Press Freedom: General freedom of speech and press.
20. Academic and Cultural Freedom: Freedom in academic and cultural activities.
21. Freedom of Discussion Men/Women: Freedom of discussion for men and women.
22. Judicial Fair Trial: Fairness in judicial trials.
23. Judicial Accountability: Accountability of the judiciary.
24. Judicial Corruption: Corruption within the judiciary.
25. Justice Access Women/Men: Access to justice for women and men.
26. Multiparty Elections: Existence of multiparty elections.
27. Inclusive Suffrage: Inclusiveness of the voting process.
28. Competitive Elections: Competitiveness of elections.
29. Free and Fair Elections: Whether elections are free and fair.
30. Autonomy of Electoral Body: Independence of the electoral body.
31. Capacity of Electoral Body: Effectiveness of the electoral body.

**Social Development Variables**: These variables relate to the overall social development and well-being of a society:
1. Health Equality: Equality in access to healthcare.
2. Education Equality: Equality in access to education.
3. Literacy Index: Literacy rates and education levels.
4. Life Expectancy: Average life expectancy.
5. Infant Mortality: Rates of infant mortality.

Based on the research questions, the following will be the key variables for analysis;

**Independent Variables (Democracy-related):**
1. Democracy Index
2. Functioning Government
3. Corruption
4. Judiciary Independence
5. Freedom of Domestic Movement for Men/Women
6. Freedom of Foreign Movement
7. Freedom of Assembly and Association
8. Speech and Press Freedom
9. Judicial Fair Trial
10. Multiparty Elections
11. Free and Fair Elections

    
**Dependent Variables (Social Development-related):**
1. Health Equality
2. Education Equality
3. Literacy Index
4. Life Expectancy
5. Infant Mortality
   
**Data Analysis Methods**
1. Descriptive Statistics: A summary of descriptive statistics for both sets of variables.
2. Correlation Analysis: The correlation between democratic variables and social development variables.
3. Regression Analysis: Regression analysis to quantify the impact of democratic variables on social development outcomes.
4. Comparative Analysis: Comparing the variations in democracy affect on social development in different regions or countries.
5. Time-Series Analysis: Trends over time.

**I NOW EXPORT THE DATAFRAME TO TABLEAU FOR FURTHER ANALYSIS**



1. DESCRIPTIVE STATISTICS

In [1]:
import pandas as pd

In [3]:
df_democracyIndex_clean=pd.read_csv('df_democracyIndex_clean.csv')


In [8]:
#Summary Statistics of numeric columns only;


# Filtering the DataFrame to exclude non-numeric columns, 'year', and 'country_code'
numeric_cols = df_democracyIndex_clean.select_dtypes(include=[float, int]).columns.tolist()
filtered_cols = [col for col in numeric_cols if col not in ['year', 'country_code']]


#Groupping by 'country' and then getting the descriptive statistics for filtered numeric variables
country_descriptive_statistics = df_democracyIndex_clean.groupby('country')[filtered_cols].describe()
country_descriptive_statistics

democracy_index                                               \
                      count      mean       std  min  25%  50%  75%  max   
country                                                                    
Afghanistan            46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
Albania                46.0  3.326087  1.367144  2.0  2.0  3.0  5.0  5.0   
Algeria                46.0  4.978261  0.147442  4.0  5.0  5.0  5.0  5.0   
Angola                 46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
Argentina              46.0  2.456522  1.089475  2.0  2.0  2.0  2.0  5.0   
...                     ...       ...       ...  ...  ...  ...  ...  ...   
Venezuela              46.0  2.826087  1.039323  2.0  2.0  2.0  4.0  5.0   
Viet Nam               46.0  5.000000  0.000000  5.0  5.0  5.0  5.0  5.0   
Yemen                  46.0  4.956522  0.206185  4.0  5.0  5.0  5.0  5.0   
Zambia                 46.0  3.673913  1.096547  2.0  3.0  3.0  5.0  5.0   
Zimbabwe               46.0  4.586957  0.497821  4.0  4.0  5.0  5.0  5.0   

            participation_interest_grps            ...  \
                                  count      mean  ...   
country                                            ...   
Afghanistan                        46.0  0.486684  ...   
Albania                            46.0  0.455124  ...   
Algeria                            46.0  0.424176  ...   
Angola                             46.0  0.395493  ...   
Argentina                          46.0  0.470221  ...   
...                                 ...       ...  ...   
Venezuela                          46.0  0.366658  ...   
Viet Nam                           46.0  0.486684  ...   
Yemen                              46.0  0.397681  ...   
Zambia                             46.0  0.442594  ...   
Zimbabwe                           46.0  0.314877  ...   

            autonomy_of_electoral_body           capacity_of_electoral_body  \
                                   75%       max                      count   
country                                                                       
Afghanistan                   0.379789  0.445602                       46.0   
Albania                       0.509041  0.579222                       46.0   
Algeria                       0.314588  0.379865                       46.0   
Angola                        0.210083  0.327613                       46.0   
Argentina                     0.658290  0.702268                       46.0   
...                                ...       ...                        ...   
Venezuela                     0.626571  0.681581                       46.0   
Viet Nam                      0.237511  0.326693                       46.0   
Yemen                         0.346269  0.520227                       46.0   
Zambia                        0.695372  0.756666                       46.0   
Zimbabwe                      0.283175  0.473184                       46.0   

                                                                         \
                 mean       std       min       25%       50%       75%   
country                                                                   
Afghanistan  0.168005  0.188566  0.000000  0.000000  0.000000  0.376231   
Albania      0.557963  0.088410  0.402637  0.506353  0.545743  0.653590   
Algeria      0.517965  0.147527  0.000000  0.560356  0.577033  0.577033   
Angola       0.114866  0.175686  0.000000  0.000000  0.000000  0.360785   
Argentina    0.571985  0.224935  0.000000  0.642948  0.642948  0.671537   
...               ...       ...       ...       ...       ...       ...   
Venezuela    0.766943  0.086958  0.575286  0.735229  0.763739  0.802891   
Viet Nam     0.761591  0.036191  0.631194  0.735546  0.788278  0.788278   
Yemen        0.231151  0.193579  0.000000  0.000000  0.318139  0.421061   
Zambia       0.295301  0.119185  0.140248  0.140248  0.353796  0.405496   
Zimbabwe     0.392395  0.092569  0.000000  0.301779 

In [9]:
# Saving the descriptive statistics as CSV for further analysis and visualization;
country_descriptive_statistics.to_csv('democracy_variables_descriptive_statistics_by_country.csv')

In [10]:
country_descriptive_statistics.columns

MultiIndex([(            'democracy_index', 'count'),
            (            'democracy_index',  'mean'),
            (            'democracy_index',   'std'),
            (            'democracy_index',   'min'),
            (            'democracy_index',   '25%'),
            (            'democracy_index',   '50%'),
            (            'democracy_index',   '75%'),
            (            'democracy_index',   'max'),
            ('participation_interest_grps', 'count'),
            ('participation_interest_grps',  'mean'),
            ...
            ( 'autonomy_of_electoral_body',   '75%'),
            ( 'autonomy_of_electoral_body',   'max'),
            ( 'capacity_of_electoral_body', 'count'),
            ( 'capacity_of_electoral_body',  'mean'),
            ( 'capacity_of_electoral_body',   'std'),
            ( 'capacity_of_electoral_body',   'min'),
            ( 'capacity_of_electoral_body',   '25%'),
            ( 'capacity_of_electoral_body',   '50%'),
            

In [13]:
#Resetting the columns to include the 'country' as a column and not an 'index-column'

#Resetting index to move 'country' from index to a regular column

country_descriptive_statistics.reset_index(inplace=True)

# Move 'country' to the first column position
columns = country_descriptive_statistics.columns.tolist()
columns.remove(('country', ''))  # Remove 'country' from its current position
new_columns = [('country', '')] + columns  # Add 'country' at the beginning
country_descriptive_statistics = country_descriptive_statistics[new_columns]

# Now, 'country' should be the first column in the DataFrame



In [14]:
country_descriptive_statistics

country level_0 index democracy_index                                \
                                         count      mean       std  min  25%   
0    Afghanistan       0     0            46.0  4.760870  0.431266  4.0  5.0   
1        Albania       1     1            46.0  3.326087  1.367144  2.0  2.0   
2        Algeria       2     2            46.0  4.978261  0.147442  4.0  5.0   
3         Angola       3     3            46.0  4.760870  0.431266  4.0  5.0   
4      Argentina       4     4            46.0  2.456522  1.089475  2.0  2.0   
..           ...     ...   ...             ...       ...       ...  ...  ...   
160    Venezuela     160   160            46.0  2.826087  1.039323  2.0  2.0   
161     Viet Nam     161   161            46.0  5.000000  0.000000  5.0  5.0   
162        Yemen     162   162            46.0  4.956522  0.206185  4.0  5.0   
163       Zambia     163   163            46.0  3.673913  1.096547  2.0  3.0   
164     Zimbabwe     164   164            46.0  4.586957  0.497821  4.0  4.0   

               ... autonomy_of_electoral_body            \
     50%  75%  ...                        75%       max   
0    5.0  5.0  ...                   0.379789  0.445602   
1    3.0  5.0  ...                   0.509041  0.579222   
2    5.0  5.0  ...                   0.314588  0.379865   
3    5.0  5.0  ...                   0.210083  0.327613   
4    2.0  2.0  ...                   0.658290  0.702268   
..   ...  ...  ...                        ...       ...   
160  2.0  4.0  ...                   0.626571  0.681581   
161  5.0  5.0  ...                   0.237511  0.326693   
162  5.0  5.0  ...                   0.346269  0.520227   
163  3.0  5.0  ...                   0.695372  0.756666   
164  5.0  5.0  ...                   0.283175  0.473184   

    capacity_of_electoral_body                                          \
                         count      mean       std       min       25%   
0                         46.0  0.168005  0.188566  0.000000  0.000000   
1                         46.0  0.557963  0.088410  0.402637  0.506353   
2                         46.0  0.517965  0.147527  0.000000  0.560356   
3                         46.0  0.114866  0.175686  0.000000  0.000000   
4                         46.0  0.571985  0.224935  0.000000  0.642948   
..                         ...       ...       ...       ...       ...   
160                       46.0  0.766943  0.086958  0.575286  0.735229   
161                       46.0  0.761591  0.036191  0.631194  0.735546   
162                       46.0  0.231151  0.193579  0.000000  0.000000   
163                       46.0  0.295301  0.119185  0.140248  0.140248   
164                       46.0  0.392395  0.092569  0.000000  0.301779   

                                   
          50%       75%       max  
0    0.000000  0.376231  0.497459  
1    0.545743  0.653590  0.653590  
2    0.577033  0.577033  0.577033  
3    0.000000  0.360785  0.392789  
4    0.642948  0.671537  0.705051  
..        ...       ...       ...  
160  0.763739  0.802891  0.964422  
161  0.788278  0.788278  0.788278  
162  0.318139  0.421061  0.474905  
163  0.353796  0.405496  0.414549  
164  0.445203  0.445203  0.493647  

[165 rows x 323 columns]

In [16]:
country_descriptive_statistics

country democracy_index                                               \
                           count      mean       std  min  25%  50%  75%  max   
0    Afghanistan            46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
1        Albania            46.0  3.326087  1.367144  2.0  2.0  3.0  5.0  5.0   
2        Algeria            46.0  4.978261  0.147442  4.0  5.0  5.0  5.0  5.0   
3         Angola            46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
4      Argentina            46.0  2.456522  1.089475  2.0  2.0  2.0  2.0  5.0   
..           ...             ...       ...       ...  ...  ...  ...  ...  ...   
160    Venezuela            46.0  2.826087  1.039323  2.0  2.0  2.0  4.0  5.0   
161     Viet Nam            46.0  5.000000  0.000000  5.0  5.0  5.0  5.0  5.0   
162        Yemen            46.0  4.956522  0.206185  4.0  5.0  5.0  5.0  5.0   
163       Zambia            46.0  3.673913  1.096547  2.0  3.0  3.0  5.0  5.0   
164     Zimbabwe            46.0  4.586957  0.497821  4.0  4.0  5.0  5.0  5.0   

    participation_interest_grps  ... autonomy_of_electoral_body            \
                          count  ...                        75%       max   
0                          46.0  ...                   0.379789  0.445602   
1                          46.0  ...                   0.509041  0.579222   
2                          46.0  ...                   0.314588  0.379865   
3                          46.0  ...                   0.210083  0.327613   
4                          46.0  ...                   0.658290  0.702268   
..                          ...  ...                        ...       ...   
160                        46.0  ...                   0.626571  0.681581   
161                        46.0  ...                   0.237511  0.326693   
162                        46.0  ...                   0.346269  0.520227   
163                        46.0  ...                   0.695372  0.756666   
164                        46.0  ...                   0.283175  0.473184   

    capacity_of_electoral_body                                          \
                         count      mean       std       min       25%   
0                         46.0  0.168005  0.188566  0.000000  0.000000   
1                         46.0  0.557963  0.088410  0.402637  0.506353   
2                         46.0  0.517965  0.147527  0.000000  0.560356   
3                         46.0  0.114866  0.175686  0.000000  0.000000   
4                         46.0  0.571985  0.224935  0.000000  0.642948   
..                         ...       ...       ...       ...       ...   
160                       46.0  0.766943  0.086958  0.575286  0.735229   
161                       46.0  0.761591  0.036191  0.631194  0.735546   
162                       46.0  0.231151  0.193579  0.000000  0.000000   
163                       46.0  0.295301  0.119185  0.140248  0.140248   
164                       46.0  0.392395  0.092569  0.000000  0.301779   

                                   
          50%       75%       max  
0    0.000000  0.376231  0.497459  
1    0.545743  0.653590  0.653590  
2    0.577033  0.577033  0.577033  
3    0.000000  0.360785  0.392789  
4    0.642948  0.671537  0.705051  
..        ...       ...       ...  
160  0.763739  0.802891  0.964422  
161  0.788278  0.788278  0.788278  
162  0.318139  0.421061  0.474905  
163  0.353796  0.405496  0.414549  
164  0.445203  0.445203  0.493647  

[165 rows x 321 columns]

In [21]:
# Reset index
country_descriptive_statistics.reset_index(drop=True, inplace=True)

# Drop the original index column if it exists
country_descriptive_statistics.drop(columns=['index'], errors='ignore', inplace=True)


C:\Users\Raphael Akeyo\AppData\Local\Temp\ipykernel_10760\3585814493.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  country_descriptive_statistics.drop(columns=['index'], errors='ignore', inplace=True)


In [22]:
country_descriptive_statistics

country democracy_index                                               \
                           count      mean       std  min  25%  50%  75%  max   
0    Afghanistan            46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
1        Albania            46.0  3.326087  1.367144  2.0  2.0  3.0  5.0  5.0   
2        Algeria            46.0  4.978261  0.147442  4.0  5.0  5.0  5.0  5.0   
3         Angola            46.0  4.760870  0.431266  4.0  5.0  5.0  5.0  5.0   
4      Argentina            46.0  2.456522  1.089475  2.0  2.0  2.0  2.0  5.0   
..           ...             ...       ...       ...  ...  ...  ...  ...  ...   
160    Venezuela            46.0  2.826087  1.039323  2.0  2.0  2.0  4.0  5.0   
161     Viet Nam            46.0  5.000000  0.000000  5.0  5.0  5.0  5.0  5.0   
162        Yemen            46.0  4.956522  0.206185  4.0  5.0  5.0  5.0  5.0   
163       Zambia            46.0  3.673913  1.096547  2.0  3.0  3.0  5.0  5.0   
164     Zimbabwe            46.0  4.586957  0.497821  4.0  4.0  5.0  5.0  5.0   

    participation_interest_grps  ... autonomy_of_electoral_body            \
                          count  ...                        75%       max   
0                          46.0  ...                   0.379789  0.445602   
1                          46.0  ...                   0.509041  0.579222   
2                          46.0  ...                   0.314588  0.379865   
3                          46.0  ...                   0.210083  0.327613   
4                          46.0  ...                   0.658290  0.702268   
..                          ...  ...                        ...       ...   
160                        46.0  ...                   0.626571  0.681581   
161                        46.0  ...                   0.237511  0.326693   
162                        46.0  ...                   0.346269  0.520227   
163                        46.0  ...                   0.695372  0.756666   
164                        46.0  ...                   0.283175  0.473184   

    capacity_of_electoral_body                                          \
                         count      mean       std       min       25%   
0                         46.0  0.168005  0.188566  0.000000  0.000000   
1                         46.0  0.557963  0.088410  0.402637  0.506353   
2                         46.0  0.517965  0.147527  0.000000  0.560356   
3                         46.0  0.114866  0.175686  0.000000  0.000000   
4                         46.0  0.571985  0.224935  0.000000  0.642948   
..                         ...       ...       ...       ...       ...   
160                       46.0  0.766943  0.086958  0.575286  0.735229   
161                       46.0  0.761591  0.036191  0.631194  0.735546   
162                       46.0  0.231151  0.193579  0.000000  0.000000   
163                       46.0  0.295301  0.119185  0.140248  0.140248   
164                       46.0  0.392395  0.092569  0.000000  0.301779   

                                   
          50%       75%       max  
0    0.000000  0.376231  0.497459  
1    0.545743  0.653590  0.653590  
2    0.577033  0.577033  0.577033  
3    0.000000  0.360785  0.392789  
4    0.642948  0.671537  0.705051  
..        ...       ...       ...  
160  0.763739  0.802891  0.964422  
161  0.788278  0.788278  0.788278  
162  0.318139  0.421061  0.474905  
163  0.353796  0.405496  0.414549  
164  0.445203  0.445203  0.493647  

[165 rows x 321 columns]

In [23]:
# Saving the descriptive statistics as CSV for further analysis and visualization;
country_descriptive_statistics.to_csv('democracy_variables_descriptive_statistics_by_country1.csv')

In [4]:
df_descriptive_statistics = pd.read_csv('democracy_variables_descriptive_statistics_by_country1.csv')

In [7]:
#The data contains 165 countries excluding regional groupings and East Germany;
#This against a total of 195 countries in the world. Comprising of 193 member countries of the United Nations and 2 observer states: the Holy See (Vatican City) and Palestine.
df_descriptive_statistics['country'].unique()

array([nan, 'Afghanistan', 'Albania', 'Algeria', 'Angola', 'Argentina',
       'Armenia', 'Australia', 'Austria', 'Azerbaijan', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Benin', 'Bolivia',
       'Bosnia and Herzegovina', 'Botswana', 'Brazil', 'Bulgaria',
       'Burkina Faso', 'Burundi', 'Cambodia', 'Cameroon', 'Canada',
       'Cape Verde', 'Central African Republic', 'Chad', 'Chile', 'China',
       'Colombia', 'Costa Rica', "Cote d'Ivoire", 'Croatia', 'Cuba',
       'Cyprus', 'Czechia', "Democratic People's Republic of Korea",
       'Democratic Republic of the Congo', 'Denmark',
       'Dominican Republic', 'Ecuador', 'Egypt', 'El Salvador',
       'Equatorial Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Fiji', 'Finland', 'France', 'Gabon', 'Gambia', 'Georgia',
       'Germany', 'Ghana', 'Greece', 'Guatemala', 'Guinea',
       'Guinea-Bissau', 'Haiti', 'Honduras', 'Hungary', 'Iceland',
       'India', 'Indonesia', 'Iran', 'Iraq', 'Ireland

**Excluded Countries**
1. Some small and closed up countries are excluded from this dataset because of: they are small and micro states/ politically sensitive and lack of data/ closed-up political systems with little or no data on democractic variables/ too-much missing values on the selected variables. 
2. The missing countries include;
Andorra/ Antigua and Barbuda/ Bahamas/ Barbados/ Belize/ Brunei Darussalam/ Cape Verde/ Comoros/ Djibouti/ Dominica/ Eswatini/ Grenada/ Guinea-Bissau/ Guyana/ Kiribati/ Liechtenstein/ Maldives/ Marshall Islands/ Monaco/ Nauru/ Palau/ Saint Kitts and Nevis/ Saint Lucia/ Saint Vincent and the Grenadines/ Samoa/ Sao Tome and Principe/ Seychelles/ Suriname/ Timor-Leste/ Tuvalu/ Vanuatu

**Categorization of the Independent Variables**

In order to do justice to the foregoing research questions, this study categorises the variables in the following manner: 
1. **Democratic Participation** (variables that seek to address issues around participation as a central tenet of democracy). They include:democratic_performance, participation_interest_grps, e_participation, freedom_assemblyandassociation, multiparty_elections, inclusive_suffrage, competetive_elections
2. **Elections and Electoral Management**: These are variables that demontrate the tenet of periodic, free and fair elections as a central principal of democracy. They include: free_fair_elections, autonomy_of_electoral_body, capacity_of_electoral_body, competetive_elections
3. **Democratic Freedoms**: freedom_of_domestic_movement_for_men, freedom_of_domestic_movement_for_women, freedom_of_foreign_movement, religious_repression, religious_freedom, freedom_assemblyandassociation, civil_society_repression, freedom_of_critisms_by_media, journalists_harrassment
speech_and_press_freedom, academic_and_cultural_freedom, freedom_of_discusion_men, freedom_of_discusion_women
4. **Accountability manifested in Corruption Index**: the variables include: corruption index, public_sector_corruption, judicial_corruption
5. **Justice and the Rule of Law**: judiciary_independence, judicial_compliance, judicial_fair_trial, judicial_accountability, judicial_corruption (overlaps with Corruption), justice_access_women, justice_access_men

**Social Development Variables**
These are the dependent variables that are used in the study to measure social development against the influence of independent democratic variables indicated above. The variables are;
1. Health Equality
2. Education Equality
3. Literacy Index
4. Life Expectancy
5. Infant Mortality


**THE FLOW OF THE WORK**

**No. 1: A Visualization of Democracy Index**
1. Mean democracy Index by Country (showing countries with strong and weak democracies cumulatively since  1975 to 2020)
2. Changes in Democracy Index per Country from 1975 to 2020

**No.2: A Visualization of the Social Development Indices**
1. Health Equality
2. Education Equality
3. Literacy Index
4. Life Expectancy
5. Infant Mortality

**No.3: Visualizations of the Relationship between the Specific Categories of Democratic Variables (Participation/ Freedoms/ Elections& Electoral Management/ Accountability(Corruption)/ Rule of Law) on each of the Social Development Indicators**
1. For the entire period of the study (1975 -2020)
2. Changes Per Country per year from 1975 to 2020

**No.4: Visualization of the Relationship between over-all Democracy Index and each of the Social Development Indicators** (Only in the weak and strong democracies to allow for comparisons)
1. For the entire period of the study (1975 -2020)
2. Changes Per Country per year from 1975 to 2020

**No.5: Story for the visualizations demonstrating trends in the influence of the democracy variables on social development indicators**

**No.6: A Dashboard combining all the analysis and visualizations of the overall relationship between democracy variables and social development indicators for all the countries from 1975 to 2020**

**-A LINK TO THE TABLEAU DASHBOARD-**

**END**

**REFRENCES**
1. Claudiu D. Tufis, Alexander Hudson, (2023). The Global State of Democracy Indices Codebook Version 7 . International IDEA Strömsborg SE–103 34 Stockholm SWEDEN
2. Huntington, S. P. (1991). Democracy's third wave. J. Democracy, 2, 12
3. pandas development team. (2023). pandas (Version 2.0) [Software]. Zenodo. https://doi.org/10.5281/zenodo.3509134
4. Python Software Foundation. (2023). Python (Version 3.10) [Software]. https://www.python.org
5. Tableau. (Version 2023.1) [Software]. (2023). Tableau Software. https://www.tableau.com
6. The International Institute for Democracy and Electoral Assistance (International IDEA). (2023). The Global State of Democracy Indices, 1975–2022, v. 7, 2023, <https://www.idea.int/gsod-indices>, (Accesed on 8th May 2024).

